lib/Lib.RandomBuffer.System.fsti

```diff
@@ -2,13 +2,40 @@ module Lib.RandomBuffer.System
 
 open Lib.IntTypes
 open Lib.Buffer
+module B = LowStar.Buffer
 
 open FStar.HyperStack.All
 
+open Lib.RandomSequence
+module R = Lib.RandomSequence
 
+/// WARNING: This function is DEPRECATED. Use ``random_crypto`` instead.
 val randombytes:
     buf: buffer uint8
   -> len: size_t{v len == length buf} ->
   Stack bool
   (requires (fun h -> live h buf))
   (ensures (fun h0 _ h1 -> modifies1 buf h0 h1))
+
+// In order to be able to reason about disjointness, we make the entropy pointer
+// recallable. Use Lib.Buffer.recall to recall that entropy_p is included
+// in the current memory snapshot.
+val entropy_p : b:lbuffer (Ghost.erased entropy) 1ul{ recallable b }
+
+/// This function waits until it has received enough entropy before
+/// generating random bytes. Note that it may wait for some time.
+val crypto_random:
+     buf: buffer uint8
+  -> len: size_t{v len == length buf} ->
+  Stack unit
+  (requires (fun h0 ->
+    live h0 buf /\
+    disjoint buf entropy_p))
+  (ensures (fun h0 ok h1 ->
```

Comment:
looks like you don't use `ok`...

Reply:
Indeed, `ok` is a unit - removed it in last commit